server/src/main/java/org/elasticsearch/script/ScriptContextStats.java

```diff
@@ -20,21 +21,33 @@
 public class ScriptContextStats implements Writeable, ToXContentFragment, Comparable<ScriptContextStats> {
     private final String context;
     private final long compilations;
+    private final TimeSeries compilationsHistory;
     private final long cacheEvictions;
+    private final TimeSeries cacheEvictionsHistory;
     private final long compilationLimitTriggered;
 
-    public ScriptContextStats(String context, long compilations, long cacheEvictions, long compilationLimitTriggered) {
+    public ScriptContextStats(String context, long compilations, long cacheEvictions, long compilationLimitTriggered,
+                              TimeSeries compilationsHistory, TimeSeries cacheEvictionsHistory) {
         this.context = Objects.requireNonNull(context);
         this.compilations = compilations;
         this.cacheEvictions = cacheEvictions;
         this.compilationLimitTriggered = compilationLimitTriggered;
+        this.compilationsHistory = compilationsHistory;
+        this.cacheEvictionsHistory = cacheEvictionsHistory;
     }
 
     public ScriptContextStats(StreamInput in) throws IOException {
         context = in.readString();
         compilations = in.readVLong();
         cacheEvictions = in.readVLong();
         compilationLimitTriggered = in.readVLong();
+        if (in.getVersion().before(Version.V_8_0_0)) {
+            compilationsHistory = null;
+            cacheEvictionsHistory = null;
+        } else {
+            compilationsHistory = in.readOptionalWriteable(TimeSeries::new);
```

Comment:
Why is this optional? I understand it could be null, but that is based on the stream version. When onOrAfter 8.0, it should always be present.

Reply:
Changed to a using a List, as there's nothing for just writing a `Writable`.  I could make these `NamedWriteable`s if you prefer.

Reply:
> there's nothing for just writing a Writable

For reading here, just construct TimeSeries:
```
compilationsHistory = new TimeSeries(in);
```

For writing, call writeTo:
```
compilationsHistory.writeTo(out);
```

Reply:
Yes, of course.  Done.